### PR TITLE
add code coverage

### PR DESCRIPTION
**Pull request by nstadelmaier-dev**
A codecov.io and codacy.com account is required for the code coverage to work since the coverage reports are uploaded to these sites.
Additionally the coverage-reporter for codacy requires a repository token (CODACY_PROJECT_TOKEN) to be present as a environment variable in the travis build environment. 

Signed-off-by: Niko <nstadelmaier.dev@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -87,6 +87,19 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>cobertura-maven-plugin</artifactId>
```
Comment (koppor) [inline pom.xml]: According to https://confluence.atlassian.com/display/CLOVER/Comparison+of+code+coverage+tools, the [JaCoCo plugin](http://www.eclemma.org/jacoco/trunk/doc/maven.html) is better then cobertura. Can you confirm? Can you switch to JaCoCo?

Comment (nstadelmaier-dev) [inline pom.xml]: JaCoCo seems to offer more functionality. It also appears to be the more active project.
I will make a switch to JaCoCo.

Comment (koppor) [inline pom.xml]: Other comparison sources: https://answers.atlassian.com/questions/9880767/clover-vs-cobertura-jacoco-and-other-code-coverage-tools

**Comment by nstadelmaier-dev** [comment]
I have made the change to JaCoCo. 
Important: The unit tests have to run! Otherwise JaCoCo does not generate a coverage report!

Problem: Some of the unit tests have to be fixed to get a valid build. Until they are fixed, travis.ci will mark every build as failed!

The following unit tests fail:
- Tests for the client (checking out the repo using SSH)
- Tests for the repository 

For detailed information run the tests local: `mvn clean install -DskipTests=false -fae`

Currently coverage upload to codacy does not work, since their uploader expects a single coverage report and report aggregation with JaCoCo is a little tricky and needs further research.
That also means that the repository token for codacy is not required at the current state.

The upload to the codecov site works and codecov automatically aggregates the uploaded reports of the modules into a coverage report viewable on the their website.

Further, I have added a parameter for either enabling or disabling tests, so that maven respects the default travis.ci install call and does not run test during dependency resolution.

Running tests can be controlled with the command line parameter ` -DskipTests`:
- ` -DskipTests=true` will run the tests
- ` -DskipTests=false` will not run the tests

Default value is `true`, so the parameter can be omitted if tests should not be run.

**Pull request merged**